rc/tools/spell.kak

```diff
@@ -148,3 +148,21 @@ define-command \
     ')
     printf 'try %%{ menu -auto-single %s }' "${menu}"
 } }
+
+define-command -params 0.. \
+    -docstring "Add the current selection to the dictionary" \
+    spell-add %{ nop %sh{
+    if [ -n "$kak_opt_spell_last_lang" ]; then
+        options="-l '$kak_opt_spell_last_lang'"
+    fi
+    if [ $# -eq 0 ]; then
+        # use selections
+        eval set -- "$kak_quoted_selections"
+    fi
+    words=""
+    while [ $# -gt 0 ]; do
+        words="$words$(printf '*%s\\n#\\n' $1)"
```

Comment:
Shouldn't the format be `'*%s\n#\n'`?

You should probably also quote `$1` as `"$1"`.

Reply:
I had to do the format string like this, otherwise the newline at the end would be removed.
If you can get it to work with that format string, be my guest

Reply:
Added the quoting

Reply:
Right, I appreciate the effort, but I don't expect performance to be much of a problem here. Maybe something like the following would make things easier (untested)?

```
    words=""
    while [ $# -gt 0 ]; do
        words="$words"$(printf '*%s\\n#\\n' "$1")
        shift
    done
    printf %b\\n "$words" | eval "aspell -a $options" 
```

↓

```
    for word; do
        if ! printf '*%s\n#\n' "${word}" | eval "aspell -a $options"; then
           printf 'fail "Unable to add word: %s"' "$(printf %s "${word}" | sed 's/"/&&/g')"
           exit 1
        fi
    done
```

Reply:
That works fine, thanks for the suggestion